examples/styling/src/main.rs

```diff
@@ -1,14 +1,35 @@
+use iced::theme::Palette;
+use iced::theme::palette::Extended;
 use iced::widget::{
     button, checkbox, column, container, horizontal_rule, progress_bar, radio,
     row, scrollable, slider, text, text_input, toggler, vertical_rule,
     vertical_space,
 };
-use iced::{Alignment, Element, Length, Sandbox, Settings, Theme};
+use iced::{Alignment, Element, Length, Sandbox, Settings, Theme, Color};
+use once_cell::sync::OnceCell;
 
 pub fn main() -> iced::Result {
+    let palette = Palette {
+        background: Color::from_rgb(1.0, 0.9, 1.0),
+        text: Color::BLACK,
+        primary: Color::from_rgb(0.5, 0.5, 0.0),
+        success: Color::from_rgb(0.0, 1.0, 0.0),
+        danger: Color::from_rgb(1.0, 0.0, 0.0),
+    };
+    let extended = Extended::generate(palette);
+    CUSTOM_THEME.set(Theme::Custom { palette, extended }).unwrap();
     Styling::run(Settings::default())
 }
 
+static CUSTOM_THEME: OnceCell<Theme> = OnceCell::new();
```

Comment:
Instead of using this static variable CUSTOM_THEME, I would instead store the custom theme in the enum ThemeType::Custom:
```rust
enum ThemeType {
    Light,
    Dark,
    Custom(Theme),
}
```
The update would be something like:
```rust
Message::ThemeChanged(theme) => self.theme = match theme {
    ThemeType::Light => Theme::Light,
    ThemeType::Dark => Theme::Dark,
    ThemeType::Custom(custom_theme) => custom_theme,
},
```